web/profiles/custom/yalesites_profile/modules/custom/ys_templated_content/src/Controller/TemplatedContentController.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace Drupal\ys_templated_content\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Session\AccountInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Controller for allowing users to create nodes based on templated content.
+ */
+class TemplatedContentController extends ControllerBase implements ContainerInjectionInterface {
+
+  /**
+   * Constructs the controller object.
+   */
+  public function __construct(AccountInterface $current_user) {
+    $this->currentUser = $current_user;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static($container->get('current_user'));
+  }
+
+  /**
+   * Builds the content selection form.
+   */
+  public function build() {
+    $formBuilder = $this->formBuilder();
+
+    return $formBuilder->getForm('Drupal\ys_templated_content\Form\TemplatedContentForm');
```

Comment:
Curious why the extra call to get a form - you could put a form in this controller itself - i.e. the way that `ys_alert` works?

Reply:
@codechefmarc I was going off of documentation I read on Drupal.org.  Are you talking about on the config side of ys_alert where it specifies the _form in the routing?  I'd love to consolidate since the Controller isn't doing much at all.

Reply:
I basically removed the controller and am directly using the form.  It seems to be fine and eliminated a file for us to manage.